### PR TITLE
Convert

### PR DESCRIPTION
**Pull request by AmirSavand**
# Converts

- [x] Link
- [x] Code
- [x] Bold
- [x] Italic
- [x] Strike
- [x] Underline
- [x] Emoji
  - [x] :+1: 
  - [x] :-1:
  - [x] :heart:

## Note

@Alir3z4 I decided markdown is totally unnecessary. We make out own and only write what we need.

We'll be adding more emojis in another pull request.

Won't be time consuming, we'll add stuff slowly.

The emojis are not handled in the best way, the best way is to convert unicodes to images (like github).

## Screens
image image image

### CODE REVIEW

File: static/assets/app.js [hunk 1]
```diff
@@ -293,6 +333,26 @@ app.controller("IndexController", function (UTILS, SETTING, VIEW, Room, Message,
     }
   };
 
+  /**
+   * Handle emojis
+   */
+  $scope.$on("whisper.Room:message", function (room, message) {
+    $timeout(function () {
```
Comment (AmirSavand) [inline static/assets/app.js]: `$timeout` without time parameter is used to run function after template has rendered. It's an AngularJS thing.

File: static/assets/app.js [hunk 2]
```diff
@@ -27,16 +27,58 @@ app.constant("UTILS", {
   }
 });
 
+/**
+ * Convert factgory
+ */
+app.factory("Convert", function ($sce) {
```
Comment (AmirSavand) [inline static/assets/app.js]: All the markup stuff we need from markdown is here which is unnecessary to load an entire library for it.

File: static/assets/app.js [hunk 3]
```diff
@@ -27,16 +27,58 @@ app.constant("UTILS", {
...
+  return {
+    link: function (text) {
+      return text.linkify();
+    },
+    bold: function (text) {
+      return text.replace(/\*(.*?)\*/g, "<b>$1</b>");
```
Comment (AmirSavand) [inline static/assets/app.js]: `*bold*`

**Comment by Alir3z4** [comment]
Why not just convert them to Unicode?

**Comment by AmirSavand** [comment]
What you mean? no one does that, they convert Unicodes into images.

**Comment by AmirSavand** [comment]
I used microsoft emojis btw.

**Comment by Alir3z4** [comment]
Ok, my bad.
I got it now

**Comment by Alir3z4** [comment]
This is good.
Let's get it in

**Comment by AmirSavand** [comment]
Awesome

**Pull request merged**